internal/process/defunct_processes.go

```diff
@@ -16,50 +16,60 @@ const ProcessFS = "/proc"
 
 // Stat represents status information of a process from /proc/[pid]/stat.
 type Stat struct {
+	// Pid is the PID of the process
+	Pid int
+
 	// Comm is the command name (usually the executable filename).
 	Comm string
 
 	// State is the state of the process.
 	State string
+
+	// PPid is the parent PID of the process
+	PPid int
 }
 
-// DefunctProcesses returns the number of zombie processes in the node.
-func DefunctProcesses() (defunctCount uint, retErr error) {
-	return DefunctProcessesForPath(ProcessFS)
+// ParseDefunctProcesses returns the number of defunct processes on the node,
+// as well as the number of defunct children of the current running process.
+func ParseDefunctProcesses() (defunctCount uint, defunctChildren []int, retErr error) {
+	return ParseDefunctProcessesForPathAndParent(ProcessFS, os.Getpid())
 }
 
-// DefunctProcessesForPath retrieves the number of zombie processes from
-// a specific process filesystem.
-func DefunctProcessesForPath(path string) (defunctCount uint, retErr error) {
+// ParseDefunctProcessesForPath retrieves the number of zombie processes from
+// a specific process filesystem, as well as the number of defunct children of the current running process.
+func ParseDefunctProcessesForPathAndParent(path string, parent int) (defunctCount uint, defunctChildren []int, retErr error) {
 	directories, err := os.Open(path)
 	if err != nil {
-		return 0, err
+		return 0, defunctChildren, err
```

Comment:
Maybe something like
```go
retErr = err
return
```
would look better here.

Reply:
hm I don't think I am a fan of that style